wire-ios-data-model/Source/MLS/MLSService.swift

```diff
@@ -355,13 +355,16 @@ public final class MLSService: MLSServiceInterface {
         ) { [weak self] _ in
             guard
                 let self,
-                let context = context,
-                ZMUser.selfUser(in: context).selfClient()?.hasRegisteredMLSClient == true else {
-                self?.logger.info("Skip periodic key material check since MLS is not enabled")
+                let context = context else {
                 return
             }
 
-            Task {
+            Task { [context] in
+                guard await context.perform({ ZMUser.selfUser(in: context).selfClient()?.hasRegisteredMLSClient == true }) == true else {
```

Comment:
```suggestion
                let hasRegisteredMLSClient = await context.perform { ZMUser.selfUser(in: context).selfClient()?.hasRegisteredMLSClient == true }

                guard hasRegisteredMLSClient else {
```
The outer `== true` is not needed anymore and I think it's easier to read.